app/src/lib/stores/app-store.ts

```diff
@@ -5837,13 +5850,70 @@ export class AppStore extends TypedBaseStore<IAppState> {
       revisionRange = revRangeInclusive(commits[0].sha, lastCommit.sha)
     }
 
-    const result = await gitStore.performFailableOperation(() =>
+    const gitStore = this.gitStoreCache.get(repository)
+    result = await gitStore.performFailableOperation(() =>
       cherryPick(repository, revisionRange, progressCallback)
     )
 
     return result || CherryPickResult.Error
   }
 
+  /**
+   * Checks for uncommitted changes before cherry pick
+   *
+   * If uncommitted changes exist, ask user to stash and return
+   * CherryPickResult.UnableToStart.
+   *
+   * If no uncommitted changes, return null.
+   */
+  private checkForUncommittedChangesBeforeCherryPick(
+    repository: Repository,
+    targetBranch: Branch,
+    commits: ReadonlyArray<CommitOneLine>
+  ): CherryPickResult | null {
+    const { changesState } = this.repositoryStateCache.get(repository)
+    const hasChanges = changesState.workingDirectory.files.length > 0
+    if (!hasChanges) {
+      return null
+    }
+
+    this._showPopup({
+      type: PopupType.LocalChangesOverwritten,
+      repository,
+      retryAction: {
+        type: RetryActionType.CherryPick,
+        repository,
+        targetBranch,
+        commits,
+      },
+      files: changesState.workingDirectory.files.map(f => f.path),
+    })
+
+    return CherryPickResult.UnableToStart
+  }
+
+  /**
+   * Attempts to checkout target branch of cherry pick operation
+   *
+   * If unable to checkout, return CherryPickResult.UnableToStart
+   * Otherwise, return null.
+   */
+  private async checkoutTargetBranchForCherryPick(
+    repository: Repository,
+    targetBranch: Branch
+  ): Promise<CherryPickResult | null> {
+    const gitStore = this.gitStoreCache.get(repository)
+
+    let checkoutSuccessful: boolean | undefined
+    await this.withAuthenticatingUser(repository, async (r, account) => {
+      checkoutSuccessful = await gitStore.performFailableOperation(() =>
+        checkoutBranch(repository, account, targetBranch)
+      )
+    })
+
+    return checkoutSuccessful ? null : CherryPickResult.UnableToStart
```

Comment:
Since `checkoutSuccessful` can also be `undefined`, we should check for `=== true` (even though it's the same in the end 😆 ).
```suggestion
    return checkoutSuccessful === true ? null : CherryPickResult.UnableToStart
```